app/routes.py

```diff
@@ -1 +1,318 @@
+        tasks = Task.query.all()
+
+    for each_task in tasks:
+        response.append(each_task.task_to_dict())
+
+    return jsonify(response), 200
+
+
+#Mark InComplete
+@task_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete_task(task_id):
+    task = validate_task(Task, task_id)
+
+    task = Task.query.get(task_id)
+    
+    #sets the task to incomplete regardless if it's complete or incomplete
+    task.completed_at = None
+    db.session.commit()
+    
+
+    return jsonify(
+        {"task": task.task_to_dict()}
+    ), 200
+
+#Mark Complete
+@task_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete_task(task_id):
+    task = validate_task(Task, task_id)
+
+    task = Task.query.get(task_id)
+
+#sets the current date 
+    task.completed_at = datetime.now()
+    db.session.commit()
+
+    #slack start
+    # slack_token = os.environ.get("SLACK_API_TOKEN")
+    slack_message = f"Somone just completed the task {task.title}"
+    # slack_client = WebClient(token=slack_token) #carries the token 
+
+    url = "https://slack.com/api/chat.postMessage"
+    slack_token = os.environ.get("SLACK_API_TOKEN")
+    data = {
+        "channel": "task-notifications",
+        "text": f"Someone just completed a task {task.title}",
+        "token": slack_token
+    }
+    response = requests.post(url, data = data)
+
+    return jsonify({
+        "task": task.task_to_dict()
+    }), 200
+
+#START OF GOAL
+
+#CREATE
+@goal_bp.route("", methods = ["POST"])
+def create_goal():
+    request_body = request.get_json()
+
+    if "title" not in request_body:
+        return jsonify({
+            "details": "Invalid data"
+        }
+        ), 400
+    
+    new_goal = Goal.from_dict(request_body)
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return jsonify(
+        {
+            "goal": new_goal.goal_to_dict()
+        }), 201
+
+
+#GET ALL
+@goal_bp.route("", methods = ["GET"])
+def get_all_goals():
+    response = []
+
+    goals = Goal.query.all()
+
+    for each_goal in goals:
+        response.append(each_goal.goal_to_dict())
+
+    return jsonify(response), 200
+
+#GET ONE
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+
+    goal = validate_goal(Goal, goal_id)
+    
+
+    return jsonify({
+        "goal": goal.goal_to_dict()
+    }), 200
+
+#UPDATE
+@goal_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+
+    goal = validate_goal(Goal, goal_id)
+    request_data = request.get_json()
+
+    goal.title = request_data["title"]
+
+    db.session.commit()
+
+    return jsonify({
+        "goal": goal.goal_to_dict()
+    }), 200
+
+#DELETE
+@goal_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+
+    goal = validate_goal(Goal, goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return{
+        "details": 'Goal 1 "Build a habit of going outside daily" successfully deleted'
```

Comment:
This response seems to be hard-coded for one specific goal. Instead, this return value should be general enough to work for any goal. Something like `f"Goal {goal_id} \"{goal.title}\" successfully deleted"`.